common/src/main/java/org/apache/uniffle/common/ShuffleBlockInfo.java

```diff
@@ -150,4 +152,8 @@ public String toString() {
 
     return sb.toString();
   }
+
+  public synchronized void copyDataTo(ByteBuf to) {
```

Comment:
`ReplicateWrite` would cause concurrency `decode`, that's why we added synchronized `copyDataTo` here.